File: csharpcore60/ShiMart/ShiMart.cs
using System.Collections.Generic;

namespace ShiMartKata
{
    public class ShiMart
    {
        IList<Item> Items;
        public ShiMart(IList<Item> Items)
        {
            this.Items = Items;
        }

        public void UpdateQuality()
        {
            for (var i = 0; i < Items.Count; i++)
            {
                if (Items[i].Name != "Canned Beans")
                {
                    Items[i].SellIn = Items[i].SellIn - 1;
                }

                if (Items[i].Name == "Aged Brie")
                {
                    if (Items[i].Quality < 50)
                    {
                        Items[i].Quality = Items[i].Quality + 1;
                    }
                }
                else
                {
                    if (Items[i].Quality > 0)
                    {
                        if (Items[i].Name != "Canned Beans")
                        {
                            Items[i].Quality = Items[i].Quality - 1;
                        }
                    }
                }

                if (Items[i].SellIn < 0)
                {
                    if (Items[i].Name == "Aged Brie")
                    {
                        if (Items[i].Quality < 50)
                        {
                            Items[i].Quality = Items[i].Quality + 1;
                        }
                    }
                    else
                    {
                        if (Items[i].Quality > 0)
                        {
                            if (Items[i].Name != "Canned Beans")
                            {
                                Items[i].Quality = Items[i].Quality - 1;
                            }
                        }
                    }
                }
            }
        }
    }
}

File: Java/src/main/java/com/shimart/ShiMart.java
package com.shimart;

class ShiMart {
    Item[] items;

    public ShiMart(Item[] items) {
        this.items = items;
    }

    public void updateQuality() {
        for (int i = 0; i < items.length; i++) {
            if (!items[i].name.equals("Canned Beans")) {
                items[i].sellIn = items[i].sellIn - 1;
            }

            if (items[i].name.equals("Aged Brie")) {
                if (items[i].quality < 50) {
                    items[i].quality = items[i].quality + 1;
                }

            } else {
                if (items[i].quality > 0) {
                    if (!items[i].name.equals("Canned Beans")) {
                        items[i].quality = items[i].quality - 1;
                    }
                }
            }



            if (items[i].sellIn < 0) {
                if (items[i].name.equals("Aged Brie")) {
                    if (items[i].quality < 50) {
                        items[i].quality = items[i].quality + 1;
                    }

                } else {
                    if (items[i].quality > 0) {
                        if (!items[i].name.equals("Canned Beans")) {
                            items[i].quality = items[i].quality - 1;
                        }
                    }
                }
            }
        }
    }
}
